### PR TITLE
Upgrade from Chromium 114 to Chromium 115

### PR DESCRIPTION
**Pull request by cdesouza-chromium**
Resolves https://github.com/brave/brave-browser/issues/30200

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by mihaiplesa — APPROVED** [review]

**Review by kylehickinson — APPROVED** [review]
iOS++

**Review by SergeyZhukovsky — APPROVED** [review]
++ wallet core

**Review by simonhong — APPROVED** [review]
browser/themes, browser/widevine, components/brave_wayback_machine ++

File: android/java/org/chromium/chrome/browser/sync/settings/BraveManageSyncSettings.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 import androidx.preference.Preference;
 
 import org.chromium.chrome.R;
-import org.chromium.chrome.browser.ui.brave_tricks.checkbox_to_switch.CheckBoxPreference;
```
[the remark this answers is not in the document]
Comment (emerick) [reply android/java/org/chromium/chrome/browser/sync/settings/BraveManageSyncSettings.java:14]: cc: @samartnik

File: app/generated_resources.grd [hunk 2]
```diff
@@ -5983,6 +5989,18 @@ Keep your key file in a safe place. You will need it to create new versions of y
...
+      </message>
+      <message name="IDS_PASSWORD_GENERATION_HELP_TEXT_SAFETY_FIRST" desc="Autofill dropdown text describing password generation focusing on safety.">
+          Stay safer online with a strong password. It will be saved to <ph name="GOOGLE_PASSWORD_MANAGER">$1<ex>Brave Password Manager</ex></ph> for <ph name="EMAIL">$2<ex>user@gmail.com</ex></ph>.
+      </message>
+      <message name="IDS_PASSWORD_GENERATION_HELP_TEXT_TRY_SOMETHING_NEW" desc="Autofill dropdown text describing password generation with a more persuasive and reassuring tone.">
+          You can always change this password later. It will be saved to <ph name="GOOGLE_PASSWORD_MANAGER">$1<ex>Brave Password Manager</ex></ph> for <ph name="EMAIL">$2<ex>user@gmail.com</ex></ph>.
+      </message>
+      <message name="IDS_PASSWORD_GENERATION_HELP_TEXT_CONVENIENCE" desc="Autofill dropdown text describing password generation with a convenience proposition.">
+          Sign up and sign in faster when your password is automatically saved to <ph name="GOOGLE_PASSWORD_MANAGER">$1<ex>Brave Password Manager</ex></ph> for <ph name="EMAIL">$2<ex>user@gmail.com</ex></ph>.
```
Comment (bsclifton) [inline app/generated_resources.grd:6002]: We might need an issue to track this - but this is trying to get the user signing in w/ Google account. We save passwords too of course, but the comparable one we have is our Sync

File: app/generated_resources.grd [hunk 3]
```diff
@@ -5983,6 +5989,18 @@ Keep your key file in a safe place. You will need it to create new versions of y
       <message name="IDS_PASSWORD_GENERATION_PROMPT_GOOGLE_PASSWORD_MANAGER" desc="Autofill dropdown text describing password generation.">
           You won't need to remember this password. It will be saved to <ph name="GOOGLE_PASSWORD_MANAGER">$1<ex>Brave Password Manager</ex></ph> for <ph name="EMAIL">$2<ex>user@gmail.com</ex></ph>.
       </message>
+      <message name="IDS_PASSWORD_GENERATION_HELP_TEXT_TRUSTED_ADVICE" desc="Autofill dropdown text describing password generation focusing on strong security messaging.">
+          Keep this account more secure with a strong password. It will be saved to <ph name="GOOGLE_PASSWORD_MANAGER">$1<ex>Brave Password Manager</ex></ph> for <ph name="EMAIL">$2<ex>user@gmail.com</ex></ph>.
```
Comment (bsclifton) [inline app/generated_resources.grd:5993]: Same as below (RE: Brave Password Manager)

File: app/generated_resources.grd [hunk 4]
```diff
@@ -7271,16 +7294,25 @@ Keep your key file in a safe place. You will need it to create new versions of y
         Side Panel Selector
       </message>
       <message name="IDS_SIDE_PANEL_COMPANION_TITLE" desc="The name of the companion feature in the side panel combo box.">
-        Brave
+        Search
+      </message>
+      <message name="IDS_SIDE_PANEL_COMPANION_TOOLBAR_TOOLTIP" desc="The tooltip for the companion feature toolbar button.">
+        Brave Search side panel
```
Comment (bsclifton) [inline app/generated_resources.grd:7300]: Does search in the side panel work?

File: app/generated_resources.grd (hunk 2 above)
Comment (emerick) [reply app/generated_resources.grd:6002]: The password manager redesign and password generation experiments are disabled upstream, but I entered https://github.com/brave/brave-browser/issues/30993 to track that and explicitly disable them with a feature override.

File: app/generated_resources.grd (hunk 3 above)
Comment (emerick) [reply app/generated_resources.grd:5993]: The password manager redesign and password generation experiments are disabled upstream, but I entered https://github.com/brave/brave-browser/issues/30993 to track that and explicitly disable them with a feature override.

File: app/os_settings_strings.grdp [hunk 5]
```diff
@@ -4808,34 +4911,43 @@ Press an assigned switch or key to remove assignment.
...
+    Customize keyboard keys
   </message>
   <message name="IDS_SETTINGS_KEYBOARD_REMAP_KEYS_ROW_SUB_LABEL" desc="Sub-label with number of remapped keys for the Remap Keyboard Keys row.">
     {COUNT, plural,
-     =0 {No keys remapped}
-     =1 {1 remapped key}
-     other {{COUNT} remapped keys}}
+     =0 {No keys customized}
+     =1 {1 customized key}
+     other {{COUNT} customized keys}}
```
Comment (bsclifton) [inline app/os_settings_strings.grdp:4920]: cc: @fallaciousreasoning on above

File: app/generated_resources.grd (hunk 4 above)
Comment (emerick) [reply app/generated_resources.grd:7300]: From what I understand, new side panel features like search will require some amount of integration work on our side before they show up in Brave. cc: @simonhong

**Review by bsclifton — COMMENTED** [review]
Left some comments for things to check out (RE: Sync vs Google Password Manager)

Building locally so I can try it out

File: app/generated_resources.grd (hunk 3 above)
Comment (bsclifton) [reply app/generated_resources.grd:5993]: I think we should definitely check out the password manager integration! 😄 cc: @rebron 

But we also just want to make sure branding shows Brave (which this does) and mentions sync for us (instead of signing into gmail account)

Comment (bsclifton) [reply app/generated_resources.grd:5993]: We can mark these comments as resolved though, since feature is disabled 👍 Or we can consider it done w/ https://github.com/brave/brave-browser/issues/30993

File: app/generated_resources.grd (hunk 2 above)
Comment (bsclifton) [reply app/generated_resources.grd:6002]: Marking as resolved - we can track with the other convo 😄

**Review by bsclifton — APPROVED** [review]
js_deps and string reviewers LGTM

File: app/generated_resources.grd (hunk 4 above)
Comment (bsclifton) [reply app/generated_resources.grd:7300]: Yes - you are correct. I built/verified we have that turned off and even if you turn on the flag (brave://flags, search for Side Search) it doesn't work 👍

File: browser/widevine/widevine_permission_request_browsertest.cc [hunk 6]
```diff
@@ -256,13 +256,17 @@ IN_PROC_BROWSER_TEST_F(ScriptTriggerWidevinePermissionRequestBrowserTest,
...
+      "supportedConfigurations.\"\n";
+
   const std::string drm_js =
       "var config = [{initDataTypes: ['cenc']}];"
       "navigator.requestMediaKeySystemAccess($1, config);";
   const std::string widevine_js = content::JsReplace(drm_js,
                                                      "com.widevine.alpha");
 
-  EXPECT_TRUE(content::ExecJs(active_contents(), widevine_js));
+  EXPECT_EQ(js_error, content::EvalJs(active_contents(), widevine_js).error);
```
Comment (goodov) [inline browser/widevine/widevine_permission_request_browsertest.cc:269]: is this error also expected in Release builds with Widevine fully enabled?

File: chromium_src/services/network/resolve_host_request.cc [hunk 7]
```diff
@@ -3,10 +3,10 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#define BRAVE_START                                   \
-  if (internal_request_->GetTextResults()) {          \
-    response_client->OnTextResults(                   \
-        internal_request_->GetTextResults().value()); \
+#define BRAVE_START                                                       \
```
Comment (goodov) [inline chromium_src/services/network/resolve_host_request.cc:6]: the define name is too generic. let's rename it:
`BRAVE_START` -> `BRAVE_RESOLVE_HOST_REQUEST_START`

File: app/generate_command_metadata.py [hunk 8]
```diff
@@ -54,6 +54,10 @@
     "IDC_BOOKMARK_BAR_UNTRACK_PRICE_FOR_SHOPPING_BOOKMARK",
     "IDC_CHROME_MENU",
     "IDC_MEDIA_TOOLBAR_CONTEXT_REPORT_CAST_ISSUE",
+    "IDC_AUTOFILL_MANDATORY_REAUTH",
+    "IDC_PASSWORDS_AND_AUTOFILL_MENU",
+    "IDC_SHOW_SEARCH_COMPANION",
+    "IDC_SHOW_CHROME_LABS",
```
Comment (goodov) [inline app/generate_command_metadata.py:60]: I *think* this should look like this:
```
@@ -54,6 +54,7 @@ EXCLUDE_COMMANDS = [
     "IDC_BOOKMARK_BAR_UNTRACK_PRICE_FOR_SHOPPING_BOOKMARK",
     "IDC_CHROME_MENU",
     "IDC_MEDIA_TOOLBAR_CONTEXT_REPORT_CAST_ISSUE",
+    "IDC_SHOW_SEARCH_COMPANION",

     # ChromeOS only
     "IDC_TOGGLE_REQUEST_TABLET_SITE",
@@ -168,6 +169,9 @@ EXISTING_TRANSLATIONS = {
     "IDC_OPEN_GUEST_PROFILE": "IDS_OPEN_GUEST_PROFILE",
     "IDC_SHOW_BRAVE_WEBCOMPAT_REPORTER": "IDS_SHOW_BRAVE_WEBCOMPAT_REPORTER",
     "IDC_COPY_CLEAN_LINK": "IDS_COPY_CLEAN_LINK",
+    "IDC_SHOW_CHROME_LABS": "IDS_CHROMELABS",
+    "IDC_AUTOFILL_MANDATORY_REAUTH": "IDS_AUTOFILL_MANDATORY_REAUTH_ICON_TOOLTIP",
+    "IDC_PASSWORDS_AND_AUTOFILL_MENU": "IDS_PASSWORDS_AND_AUTOFILL_MENU",
 }
```

File: components/content_settings/core/browser/brave_content_settings_pref_provider.cc [hunk 9]
```diff
@@ -675,48 +682,52 @@ void BravePrefProvider::UpdateCookieRules(ContentSettingsType content_type,
...
+              .session_model = google_sign_in_rule->metadata.session_model});
+      rules.emplace_back(CloneRule(firebase_rule.get()));
+      brave_cookie_rules_[incognito].emplace_back(
+          CloneRule(firebase_rule.get()));
     }
   }
 
   // non-pref based exceptions should go in the cookie_settings_base.cc
   // chromium_src override
-
```
[the remark this answers is not in the document]
Comment (emerick) [reply components/content_settings/core/browser/brave_content_settings_pref_provider.cc:703]: Fixed in 29cc8a2e3722afe16cb5113343163791618a2e8f.

File: chromium_src/services/network/resolve_host_request.cc (hunk 7 above)
Comment (emerick) [reply chromium_src/services/network/resolve_host_request.cc]: Fixed in c31dfff270216d3ebeca680888ebfa8bff1bdd75.

File: patches/ui-webui-resources-tools-bundle_js.py.patch [hunk 10]
```diff
@@ -0,0 +1,12 @@
+diff --git a/ui/webui/resources/tools/bundle_js.py b/ui/webui/resources/tools/bundle_js.py
+index 24a9ee401a97e..9114a71e5392b 100755
+--- a/ui/webui/resources/tools/bundle_js.py
++++ b/ui/webui/resources/tools/bundle_js.py
+@@ -291,6 +291,6 @@ def main(argv):
+                                    optimize_output['js_module_out_files'][0])
+     _update_dep_file(args.input, args, dep_file_header, manifest)
+ 
+-
```
[the remark this answers is not in the document]
Comment (emerick) [reply patches/ui-webui-resources-tools-bundle_js.py.patch]: Fixed in aecadfab1d8afa36a6a69f65d02a200478c75363.

File: app/generate_command_metadata.py (hunk 8 above)
Comment (emerick) [reply app/generate_command_metadata.py]: Thank you, I missed that! Fixed in afe8cdcf09c54f16837c76d0ef6276812c2a065d.

File: browser/ui/commander/commander_service_factory.cc [hunk 11]
```diff
@@ -30,8 +30,14 @@ CommanderService* CommanderServiceFactory::GetForBrowserContext(
 }
 
 CommanderServiceFactory::CommanderServiceFactory()
-    : ProfileKeyedServiceFactory("CommanderService",
-                                 ProfileSelections::BuildForAllProfiles()) {}
+    : ProfileKeyedServiceFactory(
+          "CommanderService",
+          ProfileSelections::Builder()
+              .WithRegular(ProfileSelection::kOwnInstance)
+              // TODO(crbug.com/1418376): Check if this service is needed in
```
[the remark this answers is not in the document]
Comment (emerick) [reply browser/ui/commander/commander_service_factory.cc]: I reinstated guest mode for `AcceleratorServiceFactory` here 6e9ebebc517da151db792ed85d4f2c9b09fc94a1 as it was causing several Brave and upstream tests to fail (related to running commands in guest windows and also vertical tabs in guest sessions).

I'll enter a separate issue to address `AcceleratorServiceFactory` and link it here and I'll put up a PR for it once 115 is merged if that's ok.

[the remark this answers is not in the document]
Comment (emerick) [reply browser/ui/commander/commander_service_factory.cc]: https://github.com/brave/brave-browser/issues/31025

File: android/java/org/chromium/chrome/browser/sync/settings/BraveManageSyncSettings.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlexeyBarabash) [reply android/java/org/chromium/chrome/browser/sync/settings/BraveManageSyncSettings.java:14]: fixed with https://github.com/brave/brave-core/pull/18882

**Review by AlexeyBarabash — APPROVED** [review]
sync ++

File: browser/widevine/widevine_permission_request_browsertest.cc (hunk 6 above)
Comment (iefremov) [reply browser/widevine/widevine_permission_request_browsertest.cc:269]: we do, b/c widevine is only dynamically downloaded after an explicit action with the bubble.

**Review by iefremov — APPROVED** [review]

File: browser/widevine/widevine_permission_request_browsertest.cc (hunk 6 above)
Comment (simonhong) [reply browser/widevine/widevine_permission_request_browsertest.cc:269]: yeah, we'll get this error till Widevine is downloaded and loaded.

File: chromium_src/third_party/blink/public/web/web_document.h [hunk 12]
```diff
@@ -6,7 +6,7 @@
 #ifndef BRAVE_CHROMIUM_SRC_THIRD_PARTY_BLINK_PUBLIC_WEB_WEB_DOCUMENT_H_
 #define BRAVE_CHROMIUM_SRC_THIRD_PARTY_BLINK_PUBLIC_WEB_WEB_DOCUMENT_H_
 
-#include "third_party/blink/renderer/platform/wtf/text/wtf_string.h"
+#include "third_party/blink/renderer/platform/wtf/forward.h"
```
[the remark this answers is not in the document]
Comment (goodov) [reply chromium_src/third_party/blink/public/web/web_document.h]: we should also revert additional deps/patches that were added in that incorrect fix commit:

https://github.com/brave/brave-core/pull/18388/files/26a9100ccd2dddde23b32abe5dbacaf85793de08..79836997ab19a04e8626c71501e68ab5687a86ba#diff-70c1806f0dd67e7ba587ea550ece498ec1c0700375dd127e78a539c932a39053

[the remark this answers is not in the document]
Comment (emerick) [reply chromium_src/third_party/blink/public/web/web_document.h]: Oops, I got too focused on the source code. Fixed in 825faf9bc26cc961ee7325468e7dff9b8f3c4071, thank you.

File: browser/ui/commander/commander_service_factory.cc (hunk 11 above)
[the remark this answers is not in the document]
Comment (emerick) [reply browser/ui/commander/commander_service_factory.cc]: Updated this issue to reflect that this requires some investigation to determine the appropriate course of action.

File: patches/chrome-browser-resources-settings_shared-controls-controlled_radio_button.html.patch [hunk 13]
```diff
@@ -0,0 +1,12 @@
+diff --git a/chrome/browser/resources/settings_shared/controls/controlled_radio_button.html b/chrome/browser/resources/settings_shared/controls/controlled_radio_button.html
+index b2c2c288a5fe0940f99843a4f556cfb4d73ef5bd..e9095c6c81921cd861cd5a6b1b702ee8ba5845ee 100644
+--- a/chrome/browser/resources/settings_shared/controls/controlled_radio_button.html
++++ b/chrome/browser/resources/settings_shared/controls/controlled_radio_button.html
+@@ -1,6 +1,6 @@
+ <style
+ <if expr="not chromeos_ash">
+-  include="cr-radio-button-style"
```
Comment (bridiver) [inline patches/chrome-browser-resources-settings_shared-controls-controlled_radio_button.html.patch:8]: @emerick didn't we have something similar in cr114? What did we end up doing and decided was the right fix? cc @petemill

File: patches/ui-webui-resources-tools-bundle_js.gni.patch [hunk 14]
```diff
@@ -0,0 +1,20 @@
+diff --git a/ui/webui/resources/tools/bundle_js.gni b/ui/webui/resources/tools/bundle_js.gni
+index 56dc603dc7ed6781af6961e274ec466663d9885d..3d338d333c508f05cf072c8c0d4e72512d1ba5c0 100644
+--- a/ui/webui/resources/tools/bundle_js.gni
++++ b/ui/webui/resources/tools/bundle_js.gni
+@@ -67,6 +67,7 @@ template("bundle_js") {
+         rebase_path("$root_gen_dir/ui/webui/resources/tsc/", root_build_dir)
+     is_chrome_untrusted =
+         string_replace(invoker.host, "chrome-untrusted://", "") != invoker.host
++    brave_resources_path = rebase_path("$root_gen_dir/brave/ui/webui/resources/tsc/", root_build_dir)
```
Comment (bridiver) [inline patches/ui-webui-resources-tools-bundle_js.gni.patch:9]: this should go on the same line as https://github.com/brave/brave-core/pull/18388/commits/d59b0636edd0ce39d2f82f115df4daa21d1883c0#diff-d3ec102d1b14364a2e583ead168a85b3d92f9f37e738793211db868e4c4a8385R17

Comment (bridiver) [reply patches/ui-webui-resources-tools-bundle_js.gni.patch:9]: but can we modify the node template instead so we only do this in one place?

File: patches/ui-webui-resources-tools-optimize_webui.gni.patch [hunk 15]
```diff
@@ -10,11 +10,11 @@ index 697a4526bbe325d72890d9cdecdde5e01f32b042..1344c74bcee1ec89f031823a65064ca6
  
      external_paths = []
      if (is_chromeos_ash) {
-@@ -86,6 +87,7 @@ template("optimize_webui") {
+@@ -83,6 +84,7 @@ template("optimize_webui") {
+     external_paths += [ "//resources/polymer/v3_0/|$polymer_path" ]
+ 
+     if (!is_chrome_untrusted) {
++      external_paths += [ "chrome://resources/brave/|$brave_resources_path", "//resources/brave/|$brave_resources_path" ]
```
Comment (bridiver) [inline patches/ui-webui-resources-tools-optimize_webui.gni.patch:17]: see above, but at the very least this should also collapse the patch into one line

File: patches/chrome-browser-resources-settings_shared-controls-controlled_radio_button.html.patch (hunk 13 above)
Comment (cdesouza-chromium) [reply patches/chrome-browser-resources-settings_shared-controls-controlled_radio_button.html.patch:8]: Not really. I was possibly just trying to fix conflict resolution.

File: patches/chrome-android-java-src-org-chromium-chrome-browser-sync-settings-ManageSyncSettings.java.patch [hunk 16]
```diff
@@ -1,7 +1,16 @@
...
-index 3b229431fc69c56bcbd4163ae46285a489882bf2..ee4d68cbdf2f0b83a02bb11e0429cf03462fb046 100644
+index 3b229431fc69c56bcbd4163ae46285a489882bf2..028279ee2599af11a538234a5a28df0bddad02f2 100644
 --- a/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/ManageSyncSettings.java
 +++ b/chrome/android/java/src/org/chromium/chrome/browser/sync/settings/ManageSyncSettings.java
+@@ -53,7 +53,7 @@ import org.chromium.chrome.browser.sync.ui.PassphraseDialogFragment;
+ import org.chromium.chrome.browser.sync.ui.PassphraseTypeDialogFragment;
+ import org.chromium.chrome.browser.ui.signin.SignOutDialogCoordinator;
+ import org.chromium.chrome.browser.ui.signin.SignOutDialogCoordinator.Listener;
+-import org.chromium.components.browser_ui.settings.ChromeBaseCheckBoxPreference;
++import org.chromium.components.browser_ui.settings.brave_tricks.checkbox_to_switch.ChromeBaseCheckBoxPreference;
```
Comment (bridiver) [inline patches/chrome-android-java-src-org-chromium-chrome-browser-sync-settings-ManageSyncSettings.java.patch:10]: why is this import needed?

Comment (bridiver) [inline patches/chrome-android-java-src-org-chromium-chrome-browser-sync-settings-ManageSyncSettings.java.patch:10]: we can't do this with asm?

Comment (bridiver) [reply patches/chrome-android-java-src-org-chromium-chrome-browser-sync-settings-ManageSyncSettings.java.patch:10]: change `ChromeBaseCheckBoxPreference` -> `org.chromium.components.browser_ui.settings.brave_tricks.checkbox_to_switch.ChromeBaseCheckBoxPreference`

File: patches/ui-webui-resources-tools-optimize_webui.gni.patch (hunk 15 above)
Comment (emerick) [reply patches/ui-webui-resources-tools-optimize_webui.gni.patch]: Condensed into one line here db0ffa4edcb6bc439bba91451d03dbef422bacd8. `optimize_webui` is being replaced by `bundle_js` soon, so I think the duplication will naturally go away shortly (let me know if that sounds OK).

File: patches/chrome-android-java-src-org-chromium-chrome-browser-sync-settings-ManageSyncSettings.java.patch (hunk 16 above)
Comment (bridiver) [reply patches/chrome-android-java-src-org-chromium-chrome-browser-sync-settings-ManageSyncSettings.java.patch:10]: or better yet, use asm to change the superclass for `ChromeBaseCheckBoxPreference` so we don't need to patch any sources in either?

Comment (emerick) [reply patches/chrome-android-java-src-org-chromium-chrome-browser-sync-settings-ManageSyncSettings.java.patch:10]: cc: @samartnik @AlexeyBarabash

Comment (samartnik) [reply patches/chrome-android-java-src-org-chromium-chrome-browser-sync-settings-ManageSyncSettings.java.patch:10]: We need to change it in this file only cc @AlexeyBarabash.
We can extend our bytecode patching to apply patch for certain classes only, but atm there is no such ability. I'll create a follow up issue to make it possible via bytecode.

Comment (samartnik) [reply patches/chrome-android-java-src-org-chromium-chrome-browser-sync-settings-ManageSyncSettings.java.patch:10]: Created issue https://github.com/brave/brave-browser/issues/31048

File: patches/chrome-browser-resources-settings_shared-controls-controlled_radio_button.html.patch (hunk 13 above)
Comment (emerick) [reply patches/chrome-browser-resources-settings_shared-controls-controlled_radio_button.html.patch]: Reverted here 85cafc7096c9bda3dd6fdf0b713d39e7e0b74605.

**Review by bridiver — APPROVED** [review]
